### ISSUE TITLE
Build failures with `GENERIC FAILURE` don't show error logs

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#2687]
This is a follow-up to #2684 

Failed to build [UserDefaultMacro](https://swiftpackageindex.com/maniramezan/UserDefaultMacro) with Swift 5.6 for tvOS using Xcode 13.4 at 1.0.0.

```
========================================
Build
========================================
Selected platform:         tvOS
Swift version:             5.6
Building package at path:  $PWD/checkout
GENERIC FAILURE 5.6 tvOS
```

**Comment by finestructure** [comment]
With the upcoming changes in https://gitlab.com/finestructure/swiftpackageindex-builder/-/merge_requests/230 this will be (here for iOS and 5.8 but it's the same for tvOS and 5.6, it's just that I can't run 5.6 locally):

```
========================================
Build
========================================
Selected platform:         iOS
Swift version:             5.8
Building package at path:  $PWD/checkout
Command line invocation:
    /Applications/Xcode-14.3.1.app/Contents/Developer/usr/bin/xcodebuild -IDEClonedSourcePackagesDirPathOverride=$PWD/checkout/.dependencies -resolvePackageDependencies
User defaults from command line:
    IDEClonedSourcePackagesDirPathOverride = $PWD/checkout/.dependencies
    IDEPackageSupportUseBuiltinSCM = YES
Resolve Package Graph
package 'checkout' is using Swift tools version 5.9.0 but the installed version is 5.8.0
Resolved source packages:
  checkout: $PWD/checkout
Resolve Package Graph
package 'checkout' is using Swift tools version 5.9.0 but the installed version is 5.8.0
Resolve Package Graph
package 'checkout' is using Swift tools version 5.9.0 but the installed version is 5.8.0
2023-10-31 16:37:27.782 xcodebuild[67330:12243239] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
2023-10-31 16:37:28.380 xcodebuild[67330:12243239] Writing error result bundle to /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/ResultBundle_2023-31-10_16-37-0028.xcresult
2023-10-31 16:37:29.328 xcodebuild[67330:12243239] [MT] DVTAssertions: Warning in /System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot7/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-21815/IDEFoundation/Execution/RunContextManager/IDERunContextManager.m:842
Details:  Error deleting scheme: Cannot modify data because the process disallows saving.
Object:   <IDERunContextManager: 0x60000026a370>
Method:   -deleteRunContexts:completionQueue:completionBlock:
Thread:   <_NSMainThread: 0x60000257c180>{number = 1, name = main}
Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
2023-10-31 16:37:29.329 xcodebuild[67330:12243239] could not delete old scheme: Error Domain=DVTCustomDataStoringErrors Code=0 "Cannot modify data because the process disallows saving." UserInfo={NSLocalizedDescription=Cannot modify data because the process disallows saving.}
xcodebuild: error: Could not resolve package dependencies:
  package 'checkout' is using Swift tools version 5.9.0 but the installed version is 5.8.0
Error: Shell command failed:
    env DEVELOPER_DIR=/Applications/Xcode-14.3.1.app xcrun xcodebuild -IDEClonedSourcePackagesDirPathOverride=$PWD/checkout/.dependencies -resolvePackageDependencies
```

**Comment by finestructure** [comment]
Fixed in builder 4.25.2 https://swiftpackageindex.com/builds/21BED8C0-134C-4E03-A015-AB257605310A

![CleanShot 2023-11-02 at 08 03 33@2x](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/assets/65520/935ba776-a583-46d5-af30-c5c5c2524e5c)
